[lang] [test] Improve code coverage in Matrix (#1275)
taichi-dev/taichi: docs/syntax.rst

    Currently, all functions are force-inlined. Therefore, no recursion is allowed.

.. note::

    Function arguments are passed by value.

.. note::

>     Unlike kernels, **functions do support vectors or matrices as arguments**:

Reviewer: ```suggestion
    Unlike functions, **kernels do support vectors or matrices as arguments**:
```

    .. code-block:: python

        @ti.func

>         def sdf(u):  # functions support vectors as arguments and do not need to be type-hinted

Reviewer: ```suggestion
        def sdf(u):  # functions support matrices and vectors as arguments. No type-hints needed.
```

            return u.norm() - 1

        @ti.kernel

>         def render(d_x: ti.f32, d_y: ti.f32):  # kernels doesn't, we have to use a workaround

Reviewer: ```suggestion
        def render(d_x: ti.f32, d_y: ti.f32):  # kernels do not support vector/matrix arguments yet. We have to use a workaround.
```
